client/src/main/java/org/asynchttpclient/netty/channel/DefaultChannelPool.java

```diff
@@ -257,7 +252,7 @@ private boolean offer0(Channel channel, Object partitionKey, long now) {
     }
 
     private void registerChannelCreation(Channel channel, Object partitionKey, long now) {
-        if (channelId2Creation.containsKey(partitionKey)) {
+        if (partitions.containsKey(partitionKey)) {
```

Comment:
You mean `if (!channelId2Creation.containsKey(channelId(channel))) {` right? Aka the key is not the partition key as in your sample? This plus the absence of the `!` is what got me confused initially...